modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -615,431 +646,560 @@ GDScriptParser::DataType GDScriptAnalyzer::resolve_datatype(GDScriptParser::Type
+					if (!is_type_compatible(datatype, p_member.constant->initializer->get_datatype(), true)) {
+						push_error(vformat(R"(Value of type "%s" cannot be initialized to constant of type "%s".)", p_member.constant->initializer->get_datatype().to_string(), datatype.to_string()), p_member.constant->initializer);
+					} else if (datatype.builtin_type == Variant::INT && p_member.constant->initializer->get_datatype().builtin_type == Variant::FLOAT) {
 #ifdef DEBUG_ENABLED
-							parser->push_warning(member.constant->initializer, GDScriptWarning::NARROWING_CONVERSION);
+						parser->push_warning(p_member.constant->initializer, GDScriptWarning::NARROWING_CONVERSION);
 #endif
-						}
 					}
 				}
-				datatype.is_constant = true;
+			}
+			datatype.is_constant = true;
 
-				member.constant->set_datatype(datatype);
+			p_member.constant->set_datatype(datatype);
 
-				// Apply annotations.
-				for (GDScriptParser::AnnotationNode *&E : member.constant->annotations) {
-					E->apply(parser, member.constant);
-				}
-			} break;
-			case GDScriptParser::ClassNode::Member::SIGNAL: {
-				check_class_member_name_conflict(p_class, member.signal->identifier->name, member.signal);
+			// Apply annotations.
+			for (GDScriptParser::AnnotationNode *&E : p_member.constant->annotations) {
+				E->apply(parser, p_member.constant);
+			}
+		} break;
+		case GDScriptParser::ClassNode::Member::SIGNAL: {
+			check_class_member_name_conflict(p_class, p_member.signal->identifier->name, p_member.signal);
+
+			GDScriptParser::DataType datatype;
+			datatype.kind = GDScriptParser::DataType::RESOLVING;
+			p_member.signal->set_datatype(datatype);
+
+			// Reset it after copying, to re-use.
+			datatype = GDScriptParser::DataType();
+			datatype.type_source = GDScriptParser::DataType::ANNOTATED_EXPLICIT;
+			datatype.kind = GDScriptParser::DataType::BUILTIN;
+			datatype.builtin_type = Variant::SIGNAL;
+
+			for (int j = 0; j < p_member.signal->parameters.size(); j++) {
+				GDScriptParser::DataType signal_type = resolve_datatype(p_member.signal->parameters[j]->datatype_specifier);
+				signal_type.is_meta_type = false;
+				p_member.signal->parameters[j]->set_datatype(signal_type);
+			}
 
-				for (int j = 0; j < member.signal->parameters.size(); j++) {
-					GDScriptParser::DataType signal_type = resolve_datatype(member.signal->parameters[j]->datatype_specifier);
-					signal_type.is_meta_type = false;
-					member.signal->parameters[j]->set_datatype(signal_type);
-				}
-				// TODO: Make MethodInfo from signal.
-				GDScriptParser::DataType signal_type;
-				signal_type.type_source = GDScriptParser::DataType::ANNOTATED_EXPLICIT;
-				signal_type.kind = GDScriptParser::DataType::BUILTIN;
-				signal_type.builtin_type = Variant::SIGNAL;
+			// TODO: Make MethodInfo from signal.
+			datatype.type_source = GDScriptParser::DataType::ANNOTATED_EXPLICIT;
+			datatype.kind = GDScriptParser::DataType::BUILTIN;
+			datatype.builtin_type = Variant::SIGNAL;
```

Comment:
This is duplicated code from before the `for` loop I believe.

Reply:
yes it is, fixing